qemu/tests/virtio_serial_various_chardev_hotplug.py

```diff
@@ -0,0 +1,73 @@
+from virttest import error_context
+from virttest.utils_misc import normalize_data_size
+from qemu.tests.virtio_console import add_chardev
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Hot-plug Various chardev
+
+    1. Boot a guest without chardev, no serial port & no pci
+    2. Hot plug Unix chardev backend
+    3. Hot plug udp chardev backend
+    4. Hot plug null backend
+    5. Hot plug file backend
+    6. Hot plug pty backend
+    7. Hot plug ringbuf backend
+    8. Write data to ringbuf
+    9. Read data from ringbuf
+    10. Hot-unplug Unix chardev backend
+    11. Hot-unplug udp chardev backend
+    12. Hot-unplug null backend
+    13. Hot-unplug file backend
+    14. Hot-unplug pty backend
+    15. Hot-unplug ringbuf backend
+
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    def cmd_qmp_log(vm, cmd, args):
+        reply = vm.monitor.cmd_qmp(cmd, args)
+        if "error" in reply:
+            if reply["error"]["class"] == "CommandNotFound":
+                test.cancel("qmp command %s not supported" % cmd)
+            else:
+                test.fail("qmp error: %s" % reply["error"]["desc"])
+        return reply
+
+    write_size = params.get('write_size')
+    read_size = int(params.get('read_size'))
+    if int(write_size) < read_size:
+        test.cancel("data error:write_size %d must above read_size %d"
+                    % (int(write_size), read_size))
+    normalize_data_size(write_size, "B")
```

Comment:
This is useless when you don't use the return value of `normalize_data_size`

Reply:
> This is useless when you don't use the return value of `normalize_data_size`

Just delete it, qmp will check the size automatically while hotplugging ringbuf chardev.